Drakhtar/Battalion.cpp

```diff
@@ -23,11 +50,24 @@ int Battalion::getMaxHealth() const
 
 void Battalion::loseHealth(int health)
 {
-    if (Unit::getMaxHealth() <= health)
-    {
+    if (Unit::getMaxHealth() <= health) {
         battalionSize_ -= health / Unit::getMaxHealth();
         if (battalionSize_ < 0)
             battalionSize_ = 0;
     }
     Unit::loseHealth(health);
+    sizeText_->setText("Units: " + to_string(battalionSize_));
+}
+
+void Battalion::moveToBox(Box * box)
+{
+    Unit::moveToBox(box);
+
+    sizeText_->setPos(box_->getRect().x + box_->getRect().w / 2, box_->getRect().y + box_->getRect().h * 4 / 5);
```

Comment:
You should declare `box_->getRect()` as a variable before this and re-use it instead of creating 4 objects, kind of like in my previous suggestion.